File: 1636_sort_array_by_increasing_frequency/solution.go
// LeetCode 1636 - Sort Array by Increasing Frequency
// https://leetcode.com/problems/sort-array-by-increasing-frequency/

import "sort"

func frequencySort(nums []int) []int {
	count := map[int]int{}
	for _, x := range nums {
		count[x]++
	}
	sort.Slice(nums, func(i, j int) bool {
		if count[nums[i]] == count[nums[j]] {
			return nums[i] > nums[j]
		}
		return count[nums[i]] < count[nums[j]]
	})
	return nums
}

File: 1636_sort_array_by_increasing_frequency/solution.ts
// LeetCode 1636 - Sort Array by Increasing Frequency
// https://leetcode.com/problems/sort-array-by-increasing-frequency/

function frequencySort(nums: number[]): number[] {
    const count = new Map<number, number>();
    for (const x of nums) count.set(x, (count.get(x) || 0) + 1);
    return [...nums].sort((a, b) => count.get(a)! - count.get(b)! || b - a);
}
